packages/expo-video/src/VideoView.web.tsx

```diff
@@ -50,6 +50,23 @@ export const VideoView = forwardRef((props: { player?: VideoPlayer } & VideoView
     },
   }));
 
+  useEffect(() => {
+    const fullscreenChange = () => {
+      // Emit enter if we entered fullscreen otherwise exit
+      if (document.fullscreenElement === videoRef.current) {
+        props.onFullscreenEnter?.();
+      } else {
+        props.onFullscreenExit?.();
+      }
+    };
+
+    videoRef.current?.addEventListener('fullscreenchange', fullscreenChange);
+
+    return () => {
+      videoRef.current?.removeEventListener('fullscreenchange', fullscreenChange);
+    };
+  }, [videoRef, props.onFullscreenEnter, props.onFullscreenExit]);
```

Comment:
I don't think adding a ref to a dependency array does anything, I think this useEffect code should instead be moved to the ref callback (https://github.com/expo/expo/pull/30922/files#diff-72705cdf87b0dfe8ecd5783a1ffd65f9e6257b81e046c967f4535b75b89c9835L128)
Right now it might break if ref changes, no?

Reply:
Hey @aleqsio , thanks for reviewing this. I just double-checked to be sure, and with the current state of the branch, the video screen in NCL seems to work fine in terms of this functionality. Could you clarify what you meant by moving the useEffect code to the ref callback, i.e. which ref callback are you referring to exactly?

I agree that the ref in the dependency array might be redundant - I have removed it and verified that the fullscreen events still work fine.

Reply:
I'm not sure if we're ever recreating the component without remounting the view, so it's likely the ref is going to be stable throughout the lifetime, but this might change at some point.

Look at line 127 (144 after changes) to see some additional logic done once we get a new ref (`ref={(newRef) => {...}}`) – this should trigger every time a new ref appears, so while it might be functionally the same for now I'd have more confidence it won't break in unexpected ways.

Reply:
Got it, thats what I thought. I modified the code so it triggers inside that callback. Not sure if this approach requires any cleanup code (removing the listeners) or not - if it does, I'd appreciate pointers on how to do it using this approach. Otherwise, it seems like it's good to go.

Reply:
I think it does – maybe we can assign the listener to a ref, remove the old one whenever we add a new one, and in the end, before unmount, the function will get called with a `null` value, so we can remove the last listener then.

Reply:
Hey @aleqsio, I took another stab at this and I found that the following approach works best (see video). I took the suggestion with the listener ref, but when I tried to do the event listener cleanup inside the ref callback function (on the `video` element), I was experiencing issues where I couldn't get the listeners to reliably be added/removed, especially when I would be changing the source of the player. Specifically, the listener would be properly removed when changing the source, but it wouldn't be re-set as it seems like the `if (newRef && !newRef.isEqualNode(videoRef.current)) {` check evaluates to false. This is the code that I used: 
![CleanShot 2024-08-19 at 15 44 55@2x](https://github.com/user-attachments/assets/cd7a1c45-119a-422d-b8cf-f46400d7b577)

As an alternative, I moved the setting up of the fullscreen listener to the `useEffect` where `attachAudioNodes()` and `detachAudioNodes()` is being performed, and I found it to be working perfectly (unless I'm missing something).

https://github.com/user-attachments/assets/5b398a43-6469-40cf-83f2-34daea11ed36

The console logs that you see in the above video are defined like this:
![CleanShot 2024-08-19 at 15 41 22@2x](https://github.com/user-attachments/assets/62de04a4-b502-4dc6-b0f1-b48abc1fb207)


Let me know what you think of this approach.

Reply:
If you've tested it I think I'm good with this approach

Reply:
Okay great. In that case I'm all good from my end

Reply:
Okay great. In that case I'm all good from my end